### PR TITLE
Bitbucket

### PR DESCRIPTION
**Pull request by soulseen**

### CODE REVIEW

**Comment by runzexia** [comment]
Is the bitbucket here a bitbucket server or bitbucket cloud?
Do we need to handle the two types of bitbucket separately?

**Comment by soulseen** [comment]
this is bitbucket cloud provider. yes, we also need to implement bitbucket server scm.

**Comment by runzexia** [comment]
I suggest adding some comments in the source code to indicate that bitbucket now refers to bitbucket cloud.

**Comment by soulseen** [comment]
https://github.com/kubesphere/go-scm/blob/master/scm/driver/bitbucket/bitbucket.go#L5
in top of the package, point out this is a bitbucket cloud provider.

**Comment by soulseen** [comment]
1. add bitbucket server driver
2. implements find/create/update content

/cc @runzexia

File: scm/driver/bitbucketserver/bitbucketserver.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+		io.Copy(w, res.Body)
+		return res, nil
+	}
+
+	// if a json response is expected, parse and return
+	// the json response.
+	return res, json.NewDecoder(res.Body).Decode(out)
+}
+
+// Error represents a GitLab error.
```
Comment (runzexia) [inline scm/driver/bitbucketserver/bitbucketserver.go]: bitbucket server error

File: scm/driver/bitbucketserver/content.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+func (s *contentService) Delete(ctx context.Context, repo, path, ref string) (*scm.Response, error) {
+	return nil, scm.ErrNotSupported
+}
+
+func (s *contentService) List(ctx context.Context, repo, path, ref string, opts scm.ListOptions) ([]*scm.ContentInfo, *scm.Response, error) {
+	return nil, nil, scm.ErrNotSupported
+}
+
+func stringBuffer(lines []contentLines) string {
```
Comment (runzexia) [inline scm/driver/bitbucketserver/content.go]: use string join? https://golang.org/pkg/strings/#Join

Comment (soulseen) [reply scm/driver/bitbucketserver/content.go]: If the amount of content is large, the performance will be better using this method

Comment (runzexia) [reply scm/driver/bitbucketserver/content.go]: https://github.com/golang/go/blob/master/src/strings/strings.go#L416
Look at the implementation of the official go package, there will be no performance gap

**Pull request merged**